server/src/main/java/org/elasticsearch/http/HttpChannel.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.http;
+
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.common.lease.Releasable;
+
+import java.net.InetSocketAddress;
+
+public interface HttpChannel extends Releasable {
```

Comment:
would it make sense to implement Closeable and allow to throw IOExceptions here I think it would make sense?

Reply:
Eh - they don't really throw IOExceptions as the close is being scheduled and executed on the event loop thread.